### PR TITLE
MULE-16879: create upgrade doc for 4.1 to 4.2

### PR DESCRIPTION
**Pull request by JuLuzzi**

### CODE REVIEW

File: modules/ROOT/pages/mule-runtime/updating-mule-4-versions.adoc [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+[TIP]
+You can check xref:6@studio::adding-community-runtime.adoc[this documentation page], as a reference on how to install additional runtimes on Anypoint Studio.
+
+. For every application that you update:
+.. xref:6@studio::importing-and-exporting-in-studio.adoc#exporting-a-mule-deployable-archive[Import the application] to Anypoint Studio as a Mule project.
+.. During the import remember to change the Mule version by selecting it from the *Server Runtime* dropdown menu.
++
+image::mule-runtime/updating-mule-versions-415.png[]
+
+.. Make any other changes to your Mule project that may be necessary depending on the version gap you're transitioning.
```
Comment (fsgonz) [inline modules/ROOT/pages/mule-runtime/updating-mule-4-versions.adoc]: Mule deployable archives in mule 4 are jars.

File: modules/ROOT/pages/mule-runtime/updating-mule-4-versions.adoc [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+.. xref:6@studio::importing-and-exporting-in-studio.adoc#exporting-a-mule-deployable-archive[Export your project] as a Mule Deployable Archive (.zip file).
+
+If you use Maven to build your application, edit its *pom.xml* file to change the `muleVersion` property value, so that it matches the target version.
```
Comment (fsgonz) [inline modules/ROOT/pages/mule-runtime/updating-mule-4-versions.adoc]: you always use maven to build your application in mule 4.

File: modules/ROOT/pages/mule-runtime/updating-mule-4-versions.adoc [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+Once the Mule Deployable Archives (.zip files) are in versions that you're sure are compatible with the target runtime version and you've also validated that any other necessary changes that ensure compatibility have been applied, you can move on to installing the corresponding Mule Runtime on your servers.
```
Comment (fsgonz) [inline modules/ROOT/pages/mule-runtime/updating-mule-4-versions.adoc]: jar in mule 4.

File: modules/ROOT/pages/mule-runtime/updating-mule-4-versions.adoc [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+[TIP]
+To ensure that you have faster downtime and an easier backup route, we recommend that you initially install your new Mule Runtime on a separate server from where you have your current Mule Runtime installed.
+
+. Download the new Mule version from the Customer Portal and install it (preferably on a different server or virtual machine).
+. If any of your applications use shared custom libraries, make sure you copy those .jar files to the `<MULE_HOME>/lib/user` directory of the new version.
++
+[NOTE]
+====
+If any of your previous applications rely on a full or a partial custom patch in order to work properly (these would be found in the folder `<MULE_HOME>/lib/user`), please verify that the fix is included in this target release by reviewing the corresponding xref:mule-runtime/mule-esb.adoc[release notes] page.
```
Comment (fsgonz) [inline modules/ROOT/pages/mule-runtime/updating-mule-4-versions.adoc]: lib/patches in mule 4..

File: modules/ROOT/pages/mule-runtime/updating-mule-4-versions.adoc [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+If your origin server had apps that were deployed to it through the Runtime Manager, there's no need to manually copy these, as they automatically upload to your new server when instructed to deploy them through Runtime Manager.
++
+[CAUTION]
+====
+Applications must be running and showing as `Started` in Runtime Manager. This process does not migrate stopped Applications.
+====
+
+. At this point the Mule server is already updated to the target version. Updating the Mule applications that are deployed to it is optional. If wish to do so to take full advantage of the features of the target runtime, then you must also:
+.. Follow the steps in <<Application Runtime Version>> to update your application.
+.. Find the application on the *Applications* tab on Runtime Manager and click *Choose File* to xref:runtime-manager::managing-deployed-applications.adoc#updating-your-application[upload] the new Mule Deployable Archive (.zip file).
```
Comment (fsgonz) [inline modules/ROOT/pages/mule-runtime/updating-mule-4-versions.adoc]: jar file.

**Comment by Cristian-Venticinque** [comment]
This article is being improved and migrated in PR #764

**Pull request closed without merging**